### PR TITLE
Moving to GeoIP Lite 2

### PR DESCRIPTION
**Pull request by ar2rsawseen**
Both ip4 and ip6 now have mapping to location id, and then there are separate files for location data based on this id.

So had to redo ip6 to reuse cache/buffer from ip4.

Other than that, there is no more zip code, but added eu flag and  timezone.

And for now they removed coordinates for all the data, but it seems due to client requests, they are bringing it back soon, so I just commented that one out and we can add it back when they add coordinates back.

Also attempted to fix parsing of line with grave accents, previously they failed

### CODE REVIEW

File: lib/geoip.js [hunk 1]
```diff
@@ -74,6 +74,8 @@ function lookup4(ip) {
 		range: '',
 		country: '',
 		region: '',
+        eu:'',
```
Comment (bluesmoon) [inline lib/geoip.js]: looks like indentation is off here.  tabs v/s spaces maybe

File: lib/geoip.js [hunk 2]
```diff
@@ -101,14 +103,16 @@ function lookup4(ip) {
...
-				locId = buffer.readUInt32BE((line * recordSize) + 8) - 1;
+				locId = buffer.readUInt32BE((line * recordSize) + 8);
 
 				geodata.country = locBuffer.toString('utf8', (locId * locRecordSize) + 0, (locId * locRecordSize) + 2).replace(/\u0000.*/, '');
 				geodata.region = locBuffer.toString('utf8', (locId * locRecordSize) + 2, (locId * locRecordSize) + 4).replace(/\u0000.*/, '');
-				geodata.ll = [locBuffer.readInt32BE((locId * locRecordSize) + 4) / 10000, locBuffer.readInt32BE((locId * locRecordSize) + 8) / 10000];
-				geodata.metro = locBuffer.readInt32BE((locId * locRecordSize) + 12);
-				geodata.zip = locBuffer.readInt32BE((locId * locRecordSize) + 16);
-				geodata.city = locBuffer.toString('utf8', (locId * locRecordSize) + 20, (locId * locRecordSize) + locRecordSize).replace(/\u0000.*/, '');
+                geodata.metro = locBuffer.readInt32BE((locId * locRecordSize) + 4);
```
Comment (bluesmoon) [inline lib/geoip.js]: more indentation changes.

File: lib/geoip.js [hunk 3]
```diff
@@ -131,6 +135,8 @@ function lookup4(ip) {
 function lookup6(ip) {
 	var buffer = cache6.mainBuffer;
 	var recordSize = cache6.recordSize;
+    var locBuffer = cache4.locationBuffer;
```
Comment (bluesmoon) [inline lib/geoip.js]: indent

File: package.json [hunk 4]
```diff
@@ -1,14 +1,33 @@
...
-	"version"     :  "1.2.1",
-	"description" :  "A light weight native JavaScript implementation of GeoIP API from MaxMind",
-	"keywords"    :  ["geo", "geoip", "ip", "ipv4", "ipv6", "geolookup", "maxmind", "geolite"],
-	"homepage"    :  "https://github.com/bluesmoon/node-geoip",
-	"author"      :  "Philip Tellis <philip@bluesmoon.info> (http://bluesmoon.info/)",
-	"files"       :  ["lib/", "data/", "test/","scripts/"],
-	"main"        :  "lib/geoip.js",
-	"repository"  :  { "type": "git", "url": "git://github.com/bluesmoon/node-geoip.git" },
-	"engines"     :  { "node": ">=5.10.0" },
+	"name": "geoip-lite",
```
Comment (bluesmoon) [inline package.json]: can you switch back to aligning the colons?

File: scripts/updatedb.js [hunk 5]
```diff
@@ -66,25 +67,53 @@ function mkdir(name) {
 
 // Ref: http://stackoverflow.com/questions/8493195/how-can-i-parse-a-csv-string-with-javascript
 // Return array of string values, or NULL if CSV string not well formed.
+// Return array of string values, or NULL if CSV string not well formed.
+
+function try_fixing_line(line)
+{
+    var pos1=0;
```
Comment (bluesmoon) [inline scripts/updatedb.js]: add spaces around operators and after commas (I really should have an eslint file here)

File: package.json (hunk 4 above)
Comment (ar2rsawseen) [reply package.json]: Put it back, but it is actually npm that reformats it when you use --save to add dependency, so you won't be able to hold it like that much longer :)

**Comment by ar2rsawseen** [comment]
Ok, made requested changes. And yeah, I usually use spaces instead of tabs, and in my editor and in Source tree they look exactly the same, only in github diff you can see the difference. Sorry for that ;)

**Comment by jhabdas** [comment]
Drive by. You know the right way forward. https://w3c.github.io/geolocation-sensor/

**Comment by jhabdas** [comment]
Drive by. You know the right way forward. hhttps://www.w3.org/TR/geolocation-sensor/

**Comment by jhabdas** [comment]
Drive by. You know the right way forward. https://www.w3.org/TR/geolocation-sensor/

**Comment by jhabdas** [comment]
Headless Chrome is a thing and may be run using FaaS if desired: https://github.com/adieuadieu/serverless-chrome

**Comment by ar2rsawseen** [comment]
sorry, but I really hope you are not serious 😄

**Comment by bluesmoon** [comment]
@jhabdas we'd already know the geo location of the headless chrome instance since it would be under our control. The point of geo location is to guess the location of visitors that are outside your control.

Secondly note that browser based geolocation requires the user to consent to you collecting geo data. GeoIP does not.

**Comment by jhabdas** [comment]
Good point Re: under control location.

> Secondly note that browser based geolocation requires the user to consent to you collecting geo data.

I took some time to look over MaxMind's privacy policy and that's why I wanted to raise visibility of the W3C initiative. It's my believe positioning someone in the future won't be allowed as it will become a violation of personal privacy rights. Until then I hope MaxMind and related libraries are doing more good than harm in this world.

**Comment by bluesmoon** [comment]
The only way to follow international privacy laws is to first find out which jurisdiction the user is in.

**Comment by jhabdas** [comment]
> The only way to follow international privacy laws is to first find out which jurisdiction the user is in.

You can't know that based on IP.

**Comment by ar2rsawseen** [comment]
And that is why it is not the purpose of this library, but rather user's of the library to get consents or lawful basis to use IP address to determine location

**Pull request merged**